explainer.md

```diff
@@ -178,35 +154,37 @@ function onSessionStarted(session) {
 
 ```js
 let glCanvas = document.createElement("canvas");
 let gl = glCanvas.getContext("webgl");
 
 function setupWebGLLayer() {
-  // Make sure the canvas context we want to use is compatible with the device.
-  return gl.setCompatibleXRDevice(xrDevice).then(() => {
+  // Make sure the canvas context we want to use is compatible with the current xr device.
+  return gl.makeXRCompatible().then(() => {
     // The content that will be shown on the device is defined by the session's
     // baseLayer.
     xrSession.baseLayer = new XRWebGLLayer(xrSession, gl);
   });
 }
 ```
 
-**XR Centric:** The app's primary use case is displaying XR content, and as such it doesn't mind initializing resources in an XR-centric fashion, which may include asking users to select a headset as soon as the app starts. An example would be a game which is dependent on XR presentation and input. These types of applications can avoid the need to call `setCompatibleXRDevice` and the possible context loss that it may trigger by passing the `XRDevice` that the context will be used with as a context creation argument.
+**XR Centric:** The app's primary use case is displaying XR content, and as such it doesn't mind initializing resources in an XR-centric fashion, which may include asking users to select a headset as soon as the app starts. An example would be a game which is dependent on XR presentation and input. These types of applications can avoid the need to call `makeXRCompatible` and the possible context loss that it may trigger by passing the XR device that the context will be used with as a context creation argument.
 
 ```js
-let gl = glCanvas.getContext("webgl", { compatibleXRDevice: xrDevice });
+let gl = glCanvas.getContext("webgl", { xrCompatible: true });
 ```
 
-Ensuring context compatibility with an `XRDevice` through either method may have side effects on other graphics resources in the page, such as causing the entire user agent to switch from rendering using an integrated GPU to a discrete GPU.
+Ensuring context compatibility with an XR device through either method may have side effects on other graphics resources in the page, such as causing the entire user agent to switch from rendering using an integrated GPU to a discrete GPU.
+
+If the system's underlying XR device changes (signaled by the `devicechange` event on the `navigator.xr` object) any previously set context compatibility bits will be cleared, and `makeXRCompatible` will need to be called again prior to using the context with a `XRWebGLLayer`.
```

Comment:
I know we discussed this, but I can't recall where we landed... are active `XRSession` objects supposed to end if the `devicechange` event fires?

Reply:
Ok, yes, I see that below now.  I wonder if it's worth mentioning something about it here too?